src/families/algorand/specs.js

```diff
@@ -0,0 +1,237 @@
+  const match = assetIdRegex.exec(subaccount.token.id);
+  return match !== null ? match[1] : null;
+};
+
+// Get list of ASAs associated with the account
+let getAssets = (account) => {
+  return account.subAccounts
+    ? account.subAccounts.filter((a) => a.type === "TokenAccount")
+    : [];
+};
+
+// Get list of ASAs IDs common between two accounts (intersection)
+let getCommonAssetsIds = (senderAccount, recipientAccount) => {
+  const senderAssetsIds = getAssets(senderAccount)
+    .filter((a) => a.balance.gt(0))
+    .map((a) => extractAssetId(a));
+
+  const recipientAssetsIds = getAssets(recipientAccount).map((a) =>
+    extractAssetId(a)
+  );
+
+  return senderAssetsIds.filter((assetId) =>
+    recipientAssetsIds.includes(assetId)
+  );
+};
+
+// Get Subaccount ID
+// eslint-disable-next-line no-unused-vars
+let getSubaccountId = (account, assetId) => {
+  if (account.subAccounts == null) return null;
+
+  account.subAccounts.forEach((subaccount) => {
+    if (subaccount.id.endsWith(assetId)) {
+      return subaccount.id; // e.g., libcore:1:algorand:fef9 . . . 4d7af:+algorand/asa/342836
+    }
+  });
+};
+
+const algorand: AppSpec<Transaction> = {
+  name: "Algorand",
+  currency,
+  appQuery: {
+    model: "nanoS",
+    appName: "Algorand",
+  },
+  mutations: [
+    {
+      name: "move ~50%",
+      maxRun: 2,
+      transaction: ({ account, siblings, bridge, maxSpendable }) => {
+        const spendableBalance = getSpendableBalance(maxSpendable);
+
+        const sibling = pickSiblings(siblings, 4);
+        const recipient = sibling.freshAddress;
+
+        let transaction = bridge.createTransaction(account);
+
+        let amount = spendableBalance
+          .div(1.9 + 0.2 * Math.random())
+          .integerValue();
+
+        checkSendableToEmptyAccount(amount, sibling);
+
+        const updates = [{ amount }, { recipient }];
+        return {
+          transaction,
+          updates,
+        };
+      },
+      test: ({ account, accountBeforeTransaction, operation }) => {
+        expect(account.balance.toString()).toBe(
+          accountBeforeTransaction.balance.minus(operation.value).toString()
+        );
+      },
+    },
+    {
+      name: "send max",
+      maxRun: 1,
+      transaction: ({ account, siblings, bridge, maxSpendable }) => {
+        const spendableBalance = getSpendableBalance(maxSpendable);
+        const sibling = pickSiblings(siblings, 4);
+
+        // Send the full spendable balance
+        const amount = spendableBalance;
+
+        checkSendableToEmptyAccount(amount, sibling);
+
+        return {
+          transaction: bridge.createTransaction(account),
+          updates: [
+            {
+              recipient: pickSiblings(siblings, 30).freshAddress,
+            },
+            { useAllAmount: true },
+          ],
+        };
+      },
+      test: ({ account }) => {
+        const minBalance = getMinBalance(account);
```

Comment:
correct me if i'm wrong but we shouldn't have to do this "min balance", that logic that is included in getMinBalance should be implemented as part of the account.spendableBalance. is it not the case?